Source/InterceptorStrategies.cs

```diff
@@ -133,25 +133,42 @@ public InterceptionAction HandleIntercept(ICallContext invocation, InterceptorCo
 		}
 	}
 
-	internal class InterceptToStringMixin : IInterceptStrategy
+	/// <summary>
+	/// Intercept strategy that handles `System.Object` methods.
+	/// </summary>
+	internal class InterceptObjectMethodsMixin : IInterceptStrategy
 	{
 		public InterceptionAction HandleIntercept(ICallContext invocation, InterceptorContext ctx, CurrentInterceptContext localctx)
 		{
 			var method = invocation.Method;
 
-			// Only if there is no corresponding setup
-			if (IsObjectToStringMethod(method) && !ctx.OrderedCalls.Select(c => IsObjectToStringMethod(c.Method)).Any())
+			// Only if there is no corresponding setup for `ToString()`
+			if (IsObjectMethod(method, "ToString") && !ctx.OrderedCalls.Select(c => IsObjectMethod(c.Method, "ToString")).Any())
```

Comment:
We should probably switch to using `nameof(object.ToString)` now that C# 6 is here :)